testsupport/init.go

```diff
@@ -105,6 +107,76 @@ func waitForOperators(t *testing.T) {
 	t.Log("all operators are ready and in running state")
 }
 
+func getE2EServiceAccountToken(t *testing.T) string {
+	hostNs := os.Getenv(wait.HostNsVar)
+
+	apiConfigsa, err := clientcmd.NewDefaultClientConfigLoadingRules().Load()
+	require.NoError(t, err)
+
+	// creating another config which is used for creating resclient , client, for SA
+	//so that the main kubeconfig is not altered
+	restkubeconfig, err := util.BuildKubernetesRESTConfig(*apiConfigsa)
+	require.NoError(t, err)
+
+	sacl, err := client.New(restkubeconfig, client.Options{
+		Scheme: schemeWithAllAPIs(t),
+	})
+	require.NoError(t, err)
+
+	// Check if there is already a service account present for e2e test
+	sa := &corev1.ServiceAccount{}
+	err = sacl.Get(context.TODO(), types.NamespacedName{Namespace: WaitForDeployments(t).Host().Namespace, Name: "e2e-test"}, sa)
```

Comment:
calling `WaitForDeployments` here creates a cycle
waitForOperators -> getE2EServiceAccountToken -> WaitForDeployments -> waitForOperators -> etc...
pass the host namespace as parameter as well

Reply:
nice catch, i updated it everywhere else.. but missed this one..